modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -1061,6 +1063,10 @@ class TaskProcessor {
                 task.config.exitStatus = task.exitStatus
                 task.config.errorCount = procErrCount
                 task.config.retryCount = taskErrCount
+                //Add trace of the previous execution in the task context for next execution
+                if ( traceRecord )
+                    task.config.previousTrace = traceRecord
+                task.config.previousError = error.getMessage()
```

Comment:
```suggestion
                task.config.previousException = error
```

would not make more sense expose the exception object at this point ?

Reply:
I think @bentsherman commented it https://github.com/nextflow-io/nextflow/pull/5402#discussion_r1805002690 . No sure if exposing the full object could imply a security vulnerability. I think the Exception type could be also nice to have

Reply:
The exception object can be useful for task running native java/groovy code

Reply:
Wouldn't you handle native task errors with a try / catch?

I agree security is not really an issue. I was thinking more about exposing "internal" errors to the user that are not useful or could change across releases

Reply:
Definitively not handling exception, much more for collecting error info. Exception classes are useful to understand the type of error, and specific exception brings their own info, think for example http status code, with http client exceptions

Reply:
Agreed, in nf-core you will see they typically use try-catch to handle things like http errors in helper functions

Reply:
Updated with changes in docs and test according to the modification